## Assignment 5
### Tiffany Tang
### NetID: tt2405


In [5]:
# Read in the wh data
wh <- read.csv("weight-height.csv", header = TRUE)

#Pick 20 random males and 20 random females
male_sample <- wh[1:5000, ][sample(1:5000, 20), ]
female_sample <- wh[5001:10000, ][sample(1:5000, 20), ]

In [6]:
# for height
ttest_height <- t.test(male_sample$Height, female_sample$Height)

#for weight
ttest_weight <- t.test(male_sample$Weight, female_sample$Weight)

#print results
print(ttest_height)
print(ttest_weight)


	Welch Two Sample t-test

data:  male_sample$Height and female_sample$Height
t = 5.7917, df = 37.994, p-value = 1.1e-06
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 3.370560 6.992997
sample estimates:
mean of x mean of y 
 68.82065  63.63887 


	Welch Two Sample t-test

data:  male_sample$Weight and female_sample$Weight
t = 9.9156, df = 37.917, p-value = 4.423e-12
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 40.65539 61.51658
sample estimates:
mean of x mean of y 
 187.2066  136.1206 



For both the heights and weights, the samples appear to be statistically significant to each other, as both p-values for height and weight are less than 0.05

In [7]:
# add 0.1 to each female height, 
# I will be doing this in a for loop

increment <- 0.1
curr_p_value = 1.563e-07 #from the above t test
added_value <- 0 #this is our counter

if (curr_p_value < 0.05)
    print("true")
#while the curr_p_value is < 0.05
while(curr_p_value < 0.05) {
    female_sample$Height = female_sample$Height + increment #new height with be +.1
    added_value = added_value + increment #add to our counter

    #perform t test for new height
    ttest_height <- t.test(male_sample$Height, female_sample$Height)
    curr_p_value <- ttest_height$p.value #adjust the current pvalue

    print(paste("Added:", added_value, "P-value:", curr_p_value))
    }

[1] "true"
[1] "Added: 0.1 P-value: 1.56556333895484e-06"
[1] "Added: 0.2 P-value: 2.22791242519126e-06"
[1] "Added: 0.3 P-value: 3.16941812627363e-06"
[1] "Added: 0.4 P-value: 4.50666713438602e-06"
[1] "Added: 0.5 P-value: 6.40418121837333e-06"
[1] "Added: 0.6 P-value: 9.0936580596968e-06"
[1] "Added: 0.7 P-value: 1.29006670174768e-05"
[1] "Added: 0.8 P-value: 1.82815424905393e-05"
[1] "Added: 0.9 P-value: 2.58741506491826e-05"
[1] "Added: 1 P-value: 3.65674176951959e-05"
[1] "Added: 1.1 P-value: 5.15960643127219e-05"
[1] "Added: 1.2 P-value: 7.26689621778118e-05"
[1] "Added: 1.3 P-value: 0.000102141985849072"
[1] "Added: 1.4 P-value: 0.000143249241268516"
[1] "Added: 1.5 P-value: 0.000200410154202669"
[1] "Added: 1.6 P-value: 0.000279634102212617"
[1] "Added: 1.7 P-value: 0.000389049009500765"
[1] "Added: 1.8 P-value: 0.000539585430658551"
[1] "Added: 1.9 P-value: 0.000745852816607118"
[1] "Added: 2 P-value: 0.0010272493770004"
[1] "Added: 2.1 P-value: 0.0014093504707783"
[1] "Added:

In our one case from above, the heighest we can add is 3.3 without losing significance. Adding 3.4 causes the p value to exceed 0.05 as seen above

**If you chose the heights of every female to be equal to the
mean, could you have added more before the test showed lack of significant difference ?**

Yes, if all the heights of every female equaled the mean height of the original female sample, we could add more to the female heights before the t test showed a lack of significant difference. This reduces the pooled standard deviation in the t test, making it easier to achieve a significant difference in means, even with a larger additive change to the female heights. Less variability leads to increased sensitivity in detecting differences between groups. You can add more to the female heights before the p-value exceeds 0.05 as compared to when heights retain their original variability. 


**How is the answer to question 2 related to the difference in
means between the two groups and the standard deviations of each group ?**

As you increase the heights in the females, the mean of the female group rises. This decreases the gap between the male height mean and the female height mean. Eventually the gap becomes small enough and the t test may no longer find a significantly difference leading to a p value exceeding 0.05. A smaller standard deviation in either group also leads to a smaller denominator in the t statistic formula, leading to a larger t value for the same difference in means. Therefore, being able to add a greater value to female heights is facilitated by having a sufficient difference in means and manageable standard deviations that enhance the statistical power of the test.

In [59]:
#  Order the males and females by height, pick the top 20 and bottom 20

#separate based on males and females
wh_males <- wh[1:5000, ]
wh_females <- wh[5001:10000, ]

#order the males and females by height 
wh_males_ordered <- wh_males[order(wh_males$Height), ]
wh_females_ordered <- wh_females[order(wh_females$Height), ]

#Pick top 20 (tallest) and bottom 20 (shortest) for males and females
male_tall <- wh_males_ordered[(nrow(wh_males_ordered)-19):nrow(wh_males_ordered), ]
male_short <- wh_males_ordered[1:20, ]

female_tall <- wh_females_ordered[(nrow(wh_females_ordered)-19):nrow(wh_females_ordered), ]
female_short <- wh_females_ordered[1:20, ]

#Take a look
head(male_tall)
head(male_short)
head(female_tall)
head(female_short)

Gender Height   Weight  
4303 Male   76.69096 233.1587
191  Male   76.70983 235.0354
3817 Male   76.71998 236.1467
4702 Male   76.73245 241.6866
2319 Male   76.80634 227.2203
2624 Male   76.84268 211.7242

Gender Height   Weight  
4795 Male   58.40690 121.3383
2335 Male   59.38065 136.3910
4638 Male   59.86808 117.8038
2192 Male   59.93865 141.4596
1943 Male   59.98187 112.9029
3021 Male   60.21702 146.9566

Gender Height   Weight  
5688 Female 70.66430 162.2816
7711 Female 70.66486 184.5180
8749 Female 70.69714 194.0488
5093 Female 70.71014 161.3635
7401 Female 70.74209 180.0539
9847 Female 70.75141 192.5836

Gender Height   Weight  
9286 Female 54.26313 64.70013
6625 Female 54.61686 71.39375
7295 Female 54.87373 78.60667
7618 Female 55.14856 88.81241
5346 Female 55.33649 88.36658
8693 Female 55.65189 85.62178

In [41]:
# t test between all pairs within the groups
t_test_mt_ms <- t.test(male_tall$Height, male_short$Height)  # Male-tall vs Male-short
t_test_mt_ft <- t.test(male_tall$Height, female_tall$Height)  # Male-tall vs Female-tall
t_test_mt_fs <- t.test(male_tall$Height, female_short$Height)  # Male-tall vs Female-short

t_test_ms_ft <- t.test(male_short$Height, female_tall$Height)  # Male-short vs Female-tall
t_test_ms_fs <- t.test(male_short$Height, female_short$Height)  # Male-short vs Female-short

t_test_ft_fs <- t.test(female_tall$Height, female_short$Height)  # Female-tall vs Female-short

In [42]:
# Print t-test results
print(t_test_mt_ms)
print(t_test_mt_ft)
print(t_test_mt_fs)
print(t_test_ms_ft)
print(t_test_ms_fs)
print(t_test_ft_fs)


	Welch Two Sample t-test

data:  male_tall$Height and male_short$Height
t = 77.918, df = 37.079, p-value < 2.2e-16
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 16.58676 17.47237
sample estimates:
mean of x mean of y 
 77.38604  60.35648 


	Welch Two Sample t-test

data:  male_tall$Height and female_tall$Height
t = 25.699, df = 37.998, p-value < 2.2e-16
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 5.588499 6.544245
sample estimates:
mean of x mean of y 
 77.38604  71.31967 


	Welch Two Sample t-test

data:  male_tall$Height and female_short$Height
t = 99.219, df = 36.99, p-value < 2.2e-16
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 21.17258 22.05536
sample estimates:
mean of x mean of y 
 77.38604  55.77207 


	Welch Two Sample t-test

data:  male_short$Height and female_tall$Height
t = -49.939, df = 36.992, p-value < 2.

In [43]:
#ANOVA test on the 4 groups

#first need to group all 4 categories
group_data <- data.frame(
  Height = c(male_tall$Height, male_short$Height, female_tall$Height, female_short$Height),
  Group = factor(c(rep("Male-Tall", 20), rep("Male-Short", 20), 
                   rep("Female-Tall", 20), rep("Female-Short", 20)))
)

In [44]:
# Now we can perform the test
height_result <- aov(Height ~ Group, data = group_data)
summary(height_result)

            Df Sum Sq Mean Sq F value Pr(>F)    
Group        3   5885  1961.5    4101 <2e-16 ***
Residuals   76     36     0.5                   
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

**What do you infer from the results of "a" and "b" ?**

Both the ttest results and the ANOVA results indicate a highly significant differences in height between the male and female groups, as well as between the tall and short categories within each gender. This suggests that gender and height category (tall vs. short) have substantial effects on height in this dataset. The consistency across both tests strengthens the reliability of the conclusions regarding height differences among the groups.

In [63]:
#  How many males would you have to delete from ms to make the difference not significant 
# I am going to remove males from male_short until the difference between ms and fs is >= 0.05

#male short vs female short
#initialize variables
num_to_remove <- 0
p_value <- 0
ms <- male_short[order(male_short$Height), ]
# Iteratively remove males from male-short group
while (nrow(ms) > 1) {
  # Perform the t-test
  t_test_result <- t.test(ms$Height, female_short$Height)
  p_value <- t_test_result$p.value
  
  # Check if the p-value is greater than 0.05
  if (p_value > 0.05) { #if the p value is greater than 0.05 then break from the loop
    break
  }
  
  # Remove the shortest male from the male-short group
  ms <- ms[-1, ]  # Remove the first row (the shortest)
  
  # Increment the count of removed males
  num_to_remove <- num_to_remove + 1

  print(num_to_remove)
  print(p_value)
}

[1] 1
[1] 7.25294e-24
[1] 2
[1] 1.124967e-24
[1] 3
[1] 4.229016e-24
[1] 4
[1] 7.558377e-24
[1] 5
[1] 1.502875e-23
[1] 6
[1] 4.120223e-23
[1] 7
[1] 5.520941e-23
[1] 8
[1] 8.773891e-23
[1] 9
[1] 1.907612e-22
[1] 10
[1] 1.081732e-21
[1] 11
[1] 1.259805e-20
[1] 12
[1] 1.963647e-20
[1] 13
[1] 2.460471e-20
[1] 14
[1] 3.021351e-20
[1] 15
[1] 4.709412e-20
[1] 16
[1] 6.94582e-20
[1] 17
[1] 3.070392e-19
[1] 18
[1] 2.789017e-19
[1] 19
[1] 3.651147e-19


I removed 19 of the males, and the difference between ms and fs is still significant. 
This means that even though all the shortest males were removed first, there is still a significant difference between the heights of the two groups. Removing all but one of the short males is still not enough to make the p-value between the male-short and female-short groups non-significant. This indicates that the values of the two groups are too far apart for the difference to be considered non-significant. The large difference in mean heights between the male-short and female-short groups is likely driving the statistical significance, even with only one male left in the comparisonAlternatively, removing all but one of the males drives an imbalance between the sample sizes, which could create a situation where the t-test is sensitive to any difference in the means. 

In [69]:
#male short vs male tall
num_to_remove <- 0
p_value <- 0
ms <- male_short[order(male_short$Height), ]
# Iteratively remove males from male-short group
while (nrow(ms) > 1) {
  # Perform the t-test
  t_test_result <- t.test(ms$Height, male_tall$Height)
  p_value <- t_test_result$p.value
  
  # Check if the p-value is greater than 0.05
    # If it is then exit the loop
  if (p_value > 0.05) {
    break
  }
  
  # Remove the shortest male from the male-short group
  ms <- ms[-1, ]  # Remove the first row (the shortest)
  
  # Increment the count of removed males
  num_to_remove <- num_to_remove + 1

  print(num_to_remove)
  print(p_value)
}

[1] 1
[1] 1.040749e-42
[1] 2
[1] 4.792002e-39
[1] 3
[1] 9.411578e-37
[1] 4
[1] 5.13344e-36
[1] 5
[1] 3.267959e-35
[1] 6
[1] 3.756007e-34
[1] 7
[1] 7.554031e-34
[1] 8
[1] 2.164434e-33
[1] 9
[1] 1.151717e-32
[1] 10
[1] 3.082487e-31
[1] 11
[1] 1.596792e-29
[1] 12
[1] 3.378362e-29
[1] 13
[1] 5.156414e-29
[1] 14
[1] 7.608901e-29
[1] 15
[1] 1.570103e-28
[1] 16
[1] 2.955729e-28
[1] 17
[1] 2.296525e-27
[1] 18
[1] 2.104231e-27
[1] 19
[1] 3.108876e-27


Similarly to above, removing all but one of the short males is still not enough to make the p-value between the male-short and male-tall groups non-significant. This indicates that the values of the two groups are too far apart for the difference to be considered non-significant. The large difference in mean heights between the groups is likely driving the statistical significance, even with only one short male left in the comparison

In [70]:
#male short vs female tall
num_to_remove <- 0
p_value <- 0
ms <- male_short[order(male_short$Height), ]
# Iteratively remove males from male-short group
while (nrow(ms) > 1) {
  # Perform the t-test
  t_test_result <- t.test(ms$Height, female_tall$Height)
  p_value <- t_test_result$p.value
  
  # Check if the p-value is greater than 0.05
    # If it is then exit the loop
  if (p_value > 0.05) {
    break
  }
  
  # Remove the shortest male from the male-short group
  ms <- ms[-1, ]  # Remove the first row (the shortest)
  
  # Increment the count of removed males
  num_to_remove <- num_to_remove + 1

  print(num_to_remove)
  print(p_value)
}

[1] 1
[1] 1.495856e-35
[1] 2
[1] 7.954065e-33
[1] 3
[1] 4.882524e-31
[1] 4
[1] 1.902142e-30
[1] 5
[1] 8.366059e-30
[1] 6
[1] 5.845807e-29
[1] 7
[1] 1.045066e-28
[1] 8
[1] 2.466417e-28
[1] 9
[1] 9.460386e-28
[1] 10
[1] 1.294293e-26
[1] 11
[1] 3.011378e-25
[1] 12
[1] 5.565931e-25
[1] 13
[1] 7.93041e-25
[1] 14
[1] 1.100045e-24
[1] 15
[1] 1.992769e-24
[1] 16
[1] 3.353808e-24
[1] 17
[1] 1.743829e-23
[1] 18
[1] 1.635417e-23
[1] 19
[1] 2.251637e-23


In all three cases, removing males from the male-short group was not enough to make the comparison between the male-short and female-short groups statistically non-significant. This suggests that the difference between the two groups is large enough that even reducing the male sample size significantly does not eliminate the statistical significance.

In [74]:
#ANOVA TEST
group_data <- data.frame(
  Height = c(male_tall$Height, male_short$Height, female_tall$Height, female_short$Height),
  Group = factor(c(rep("Male-Tall", 20), rep("Male-Short", 20), 
                   rep("Female-Tall", 20), rep("Female-Short", 20)))
)
#for the whole data set without any removals
anova_result <- aov(Height ~ Group, data = group_data)
summary(anova_result)

            Df Sum Sq Mean Sq F value Pr(>F)    
Group        3   5885  1961.5    4101 <2e-16 ***
Residuals   76     36     0.5                   
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

In [89]:
group_data <- data.frame(
  Height = c(male_tall$Height, male_short$Height[20:20], female_tall$Height, female_short$Height),
  Group = factor(c(rep("Male-Tall", 20), rep("Male-Short", 1), 
                   rep("Female-Tall", 20), rep("Female-Short", 20)))
)
anova_result <- aov(Height ~ Group, data = group_data)
summary(anova_result)

            Df Sum Sq Mean Sq F value Pr(>F)    
Group        3   5022  1674.0    3325 <2e-16 ***
Residuals   57     29     0.5                   
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

The ANOVA test indicates that removing males from the male-short group was insufficient to make the comparison with the male-tall, female-short, and female-tall groups statistically non-significant. This suggests that the differences between the male-short group and all other groups are large enough that even significantly reducing the male sample size does not eliminate the statistical significance.

**How many femelas would you have to delete from ft to reduce significance of the difference ? (you would delete
the tallest females first).**

In [94]:
##ANOVA test
group_data <- data.frame(
  Height = c(male_tall$Height, male_short$Height, female_tall$Height[20:20], female_short$Height),
  Group = factor(c(rep("Male-Tall", 20), rep("Male-Short", 20), 
                   rep("Female-Tall", 1), rep("Female-Short", 20)))
)
anova_result <- aov(Height ~ Group, data = group_data)
summary(anova_result)

            Df Sum Sq Mean Sq F value Pr(>F)    
Group        3   5266  1755.2    3896 <2e-16 ***
Residuals   57     26     0.5                   
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

In [101]:
test<- t.test(female_tall$Height, female_short$Height)
print(test)


	Welch Two Sample t-test

data:  female_tall$Height and female_short$Height
t = 71.051, df = 36.9, p-value < 2.2e-16
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 15.10418 15.99101
sample estimates:
mean of x mean of y 
 71.31967  55.77207 



In [106]:
# How many females would you have to delete from ft to make the difference not significant?
# Initialize variables
num_to_remove <- 0
p_value <- 0

# Sort female-tall group for removal (remove the tallest first)
ft <- female_tall[order(-female_tall$Height), ]
fs <- female_short  # Use the full female-short group

# Iteratively remove females from female-tall group
while (nrow(ft) > 1) {
  # Perform the t-test between female tall and female short
  t_test_result <- t.test(ft$Height, female_short$Height)
  p_value <- t_test_result$p.value
  
  # Check if the p-value is greater than 0.05
  if (p_value > 0.05) {
    break
  }
  
  # Remove the tallest female from the female-tall group
  cat("Removing tallest female height:", ft$Height[1], "\n")
  ft <- ft[-1, ]  # Remove the first row (the tallest)
  
  # Increment the count of removed females
  num_to_remove <- num_to_remove + 1

  print(num_to_remove)
  print(p_value)
}




Removing tallest female height: 73.38959 
[1] 1
[1] 4.439294e-41
Removing tallest female height: 72.42977 
[1] 2
[1] 6.358086e-43
Removing tallest female height: 72.26475 
[1] 3
[1] 2.239687e-42
Removing tallest female height: 72.08871 
[1] 4
[1] 1.906532e-41
Removing tallest female height: 71.99126 
[1] 5
[1] 6.562196e-40
Removing tallest female height: 71.5522 
[1] 6
[1] 8.467987e-37
Removing tallest female height: 71.43895 
[1] 7
[1] 4.549301e-35
Removing tallest female height: 71.27751 
[1] 8
[1] 3.611072e-33
Removing tallest female height: 71.12279 
[1] 9
[1] 1.037411e-31
Removing tallest female height: 71.07045 
[1] 10
[1] 6.536253e-31
Removing tallest female height: 70.95066 
[1] 11
[1] 5.841448e-30
Removing tallest female height: 70.93219 
[1] 12
[1] 1.41459e-29
Removing tallest female height: 70.8371 
[1] 13
[1] 7.020164e-29
Removing tallest female height: 70.81754 
[1] 14
[1] 1.150486e-28
Removing tallest female height: 70.75141 
[1] 15
[1] 2.804983e-28
Removing tallest femal

Still significant, even after removing all tall females- ft vs fs

In [107]:
# How many females would you have to delete from ft to make the difference not significant?
# Initialize variables
num_to_remove <- 0
p_value <- 0

# Sort female-tall group for removal (remove the tallest first)
ft <- female_tall[order(-female_tall$Height), ]

# Iteratively remove females from female-tall group
while (nrow(ft) > 1) {
  # Perform the t-test between female tall and female short
  t_test_result <- t.test(ft$Height, male_tall$Height)
  p_value <- t_test_result$p.value
  
  # Check if the p-value is greater than 0.05
  if (p_value > 0.05) {
    break
  }
  
  # Remove the tallest female from the female-tall group
  ft <- ft[-1, ]  # Remove the first row (the tallest)
  
  # Increment the count of removed females
  num_to_remove <- num_to_remove + 1

  print(num_to_remove)
  print(p_value)
}



[1] 1
[1] 1.256794e-25
[1] 2
[1] 2.113337e-26
[1] 3
[1] 3.641643e-26
[1] 4
[1] 1.077492e-25
[1] 5
[1] 6.300063e-25
[1] 6
[1] 1.879505e-23
[1] 7
[1] 9.789177e-23
[1] 8
[1] 5.977533e-22
[1] 9
[1] 2.320288e-21
[1] 10
[1] 4.710097e-21
[1] 11
[1] 1.120812e-20
[1] 12
[1] 1.53974e-20
[1] 13
[1] 2.888482e-20
[1] 14
[1] 3.410877e-20
[1] 15
[1] 4.781456e-20
[1] 16
[1] 4.80331e-20
[1] 17
[1] 5.472153e-20
[1] 18
[1] 5.528875e-20
[1] 19
[1] 6.814322e-20
Number of females removed to make the difference non-significant: 19 


still siginificant even after removing all tall females- ft vs mt

In [108]:
# How many females would you have to delete from ft to make the difference not significant?
# Initialize variables
num_to_remove <- 0
p_value <- 0

# Sort female-tall group for removal (remove the tallest first)
ft <- female_tall[order(-female_tall$Height), ]

# Iteratively remove females from female-tall group
while (nrow(ft) > 1) {
  # Perform the t-test between female tall and female short
  t_test_result <- t.test(ft$Height, male_short$Height)
  p_value <- t_test_result$p.value
  
  # Check if the p-value is greater than 0.05
  if (p_value > 0.05) {
    break
  }
  
  # Remove the tallest female from the female-tall group
  ft <- ft[-1, ]  # Remove the first row (the tallest)
  
  # Increment the count of removed females
  num_to_remove <- num_to_remove + 1

  print(num_to_remove)
  print(p_value)
}



[1] 1
[1] 1.495856e-35
[1] 2
[1] 3.101141e-37
[1] 3
[1] 8.210369e-37
[1] 4
[1] 4.533992e-36
[1] 5
[1] 8.054951e-35
[1] 6
[1] 2.885252e-32
[1] 7
[1] 7.934634e-31
[1] 8
[1] 3.044875e-29
[1] 9
[1] 5.069117e-28
[1] 10
[1] 2.391996e-27
[1] 11
[1] 1.512587e-26
[1] 12
[1] 3.210621e-26
[1] 13
[1] 1.242454e-25
[1] 14
[1] 1.898108e-25
[1] 15
[1] 4.045184e-25
[1] 16
[1] 4.370121e-25
[1] 17
[1] 6.080303e-25
[1] 18
[1] 6.605936e-25
[1] 19
[1] 1.058551e-24


The three t test and ANOVA test indicates that removing females from the female tall group was insufficient to make the comparison with the male-tall, female-short, and male-short groups statistically non-significant. This suggests that the differences between the female-tall group and all other groups are large enough that even significantly reducing the female sample size does not eliminate the statistical significance.

**Could you use the ms v/s ft, with or without deletions to prove the opposite of what we know from the full set (on average, males are taller than females).**

In [66]:
summary(male_short)
summary(female_tall)
#just to take a look

    Gender              Height          Weight     
 Length:20          Min.   :58.41   Min.   :112.9  
 Class :character   1st Qu.:60.16   1st Qu.:127.9  
 Mode  :character   Median :60.49   Median :135.5  
                    Mean   :60.36   Mean   :135.7  
                    3rd Qu.:60.80   3rd Qu.:142.8  
                    Max.   :60.97   Max.   :156.5  

    Gender              Height          Weight     
 Length:20          Min.   :70.66   Min.   :161.4  
 Class :character   1st Qu.:70.75   1st Qu.:176.0  
 Mode  :character   Median :71.01   Median :183.3  
                    Mean   :71.32   Mean   :181.4  
                    3rd Qu.:71.66   3rd Qu.:190.6  
                    Max.   :73.39   Max.   :194.0  

In [68]:
# Perform a t test between the two groups
test <- t.test(male_short$Height, female_tall$Height)
print(test)


	Welch Two Sample t-test

data:  male_short$Height and female_tall$Height
t = -49.939, df = 36.992, p-value < 2.2e-16
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -11.40801 -10.51838
sample estimates:
mean of x mean of y 
 60.35648  71.31967 



The p-value (2.2e-16) indicates a statistically significant difference between the heights of the male-short (without any deletions) and female-tall groups. The mean height of males in the male-short group is 60.36, while the mean height of females in the female-tall group is 71.32. This comparison highlights that, in this specific subset of data, the average height of short males is less than that of tall females, which contrasts with the general observation from the full dataset that, on average, males are taller than females.